zulipterminal/model.py

```diff
@@ -116,8 +116,20 @@ def __init__(self, controller: Any) -> None:
         (self.stream_dict, self.muted_streams,
          self.pinned_streams, self.unpinned_streams) = stream_data
 
+        # NOTE: The expected response has been upgraded from
+        # [stream_name, topic] to [stream_name, topic, date_muted] in
+        # 3.0, Feature level 1.
         self._muted_topics = (
-            self.initial_data['muted_topics'])  # type: List[List[str]]
+            self.initial_data['muted_topics']
+        )  # type: List[List[Union[str, int]]]
+        assert all(len(sub_list) in (2, 3) for sub_list in self._muted_topics)
+        # The intent is to use the following data structure locally to avoid
+        # server version & feature level checks throughout the codebase.
+        self._muted_topics_dict = {
+            (stream_name, topic): (None if self.server_feature_level is None
+                                   else date_muted[0])
+            for stream_name, topic, *date_muted in self._muted_topics
+        }
```

Comment:
I'm glad you understood what I was suggesting here - this looks much simpler :+1:

I'm not sure we still need to store `self._muted_topics` though? Can we just name the final variable that instead?
(I know we have `_dict` on other variables, but including the type in the name is often superfluous)